peekaboo/toolbox/ms_office.py

```diff
@@ -59,3 +59,34 @@ def has_office_macros(office_file):
     except Exception as error:
         logger.exception(error)
         return False
+
+def has_office_macros_with_auto_action(office_file):
+    """
+    Detects macros for Auto_ actions in Microsoft Office documents.
+
+    @param office_file: The MS Office document to check for Auto_ macros.
+    @return: True if Auto_ macros where found, otherwise False.
+             If VBA_Parser crashes it returns False too.
+    """
+
+    SUSPICIOUS_KEYWORDS = ["AutoOpen", "AutoClose"]
+
+    file_extension = office_file.split('.')[-1]
+    if file_extension not in MS_OFFICE_EXTENSIONS:
+        print("/%s/" % file_extension)
+        return False
+    try:
+        # VBA_Parser reports macros for office documents
+        vbaparser = VBA_Parser(office_file)
+        for (filename, stream_path, vba_filename, vba_code) in vbaparser.extract_macros():
+            for w in  SUSPICIOUS_KEYWORDS:
+                if w in vba_code:
```

Comment:
Since this is only a string match, it would also match substrings anywhere in the code, not just function names, again leading to false positives. Should we go with `analyze_macros()` as the oletools documentation suggests (https://github.com/decalage2/oletools/wiki/olevba#analyze-vba-source-code)? This seems to try to parse the code and extract function names and other keywords which we could more reliably match against. Untested:
```python
        for kw_type, keyword, description in vbaparser.extract_macros():
            if keyword in  SUSPICIOUS_KEYWORDS:
```
We could even match against kw_type being `AutoExec` to be even more precise.

Reply:
Addendum: Identifier names in VBA are not case-sensitive. So we do need to do a case-insensitive match here to find all possible spellings of `autoopen`, `AutoOpen` and `auTOoPen`. Current code:
```python
    SUSPICIOUS_KEYWORDS = ["autoopen", "autoclose"]
[...]
                if w in vba_code.lower():
```
Proposed `analyze_macros()` solution:
```python
    SUSPICIOUS_KEYWORDS = ["autoopen", "autoclose"]
[...]
        for kw_type, keyword, description in vbaparser.analyze_macros():
            if keyword.lower() in SUSPICIOUS_KEYWORDS:
```
Or treat it as a regex:
```python
        for kw_type, keyword, description in vbaparser.analyze_macros():
            for w in SUSPICIOUS_KEYWORDS:
                if re.match(w, keyword, , re.IGNORECASE):
```

Reply:
Done in 6a3ba2803d3f077b836d9755fb46e273bdb912b0.